power_grid_model_io_native_c/power_grid_model_io_native/include/power_grid_model_io_native/vnf_converter/vnf_pgm_converter.hpp

```diff
@@ -32,21 +34,22 @@ inline std::string serialize_data(power_grid_model::ConstDataset const& const_da
 
 class PgmVnfConverter {
   public:
-    PgmVnfConverter(char* buffer = nullptr, power_grid_model::WritableDataset* data = nullptr);
+    PgmVnfConverter(const char* buffer = nullptr, power_grid_model::WritableDataset* data = nullptr,
+                    power_grid_model_io_native::Idx experimental_feature_flag = 0);
```

Comment:
for your own sanity: please use a `using` declaration or make sure you're in the correct namespace.
```suggestion
                    Idx experimental_feature_flag = 0);
```

in addition, you can/should use an enum for this

Reply:
Made sure that it is in the correct namespace. However, I have a question regarding the enum. Experimental_feature enums are in basics.h, but we do not include .h files in .hpp files. Do I create my own enum representation just like in PGM's enum.hpp?

Reply:
> Do I create my own enum representation just like in PGM's enum.hpp?

yes, that's also how we do it in the PGM. the reason is that you want to keep interface as much separate from the implementation as possible. if not, you're in for a world of pain if you ever want to refactor